src/qibolab/platforms/abstract.py

```diff
@@ -325,95 +351,28 @@ def sweep(self, sequence, *sweepers, nshots=1024, average=True, relaxation_time=
         # Check in the settings if qubits[0]-qubits[1] is a key
-        if f"{qubits[0]}-{qubits[1]}" in self.settings["native_gates"]["two_qubit"]:
-            pulse_sequence_settings = self.settings["native_gates"]["two_qubit"][f"{qubits[0]}-{qubits[1]}"]["CZ"]
-        elif f"{qubits[1]}-{qubits[0]}" in self.settings["native_gates"]["two_qubit"]:
-            pulse_sequence_settings = self.settings["native_gates"]["two_qubit"][f"{qubits[1]}-{qubits[0]}"]["CZ"]
-        else:
+        if frozenset(qubits) not in self.two_qubit_natives:
             raise_error(
                 ValueError,
                 f"Calibration for CZ gate between qubits {qubits[0]} and {qubits[1]} not found.",
             )
-
-        # If settings contains only one pulse dictionary, convert it into a list that can be iterated below
-        if isinstance(pulse_sequence_settings, dict):
-            pulse_sequence_settings = [pulse_sequence_settings]
-
-        from qibolab.pulses import FluxPulse, PulseSequence
-
-        sequence = PulseSequence()
-        virtual_z_phases = defaultdict(int)
-
-        for pulse_settings in pulse_sequence_settings:
-            if pulse_settings["type"] == "qf":
-                qf_duration = pulse_settings["duration"]
-                qf_amplitude = pulse_settings["amplitude"]
-                qf_shape = pulse_settings["shape"]
-                qubit = pulse_settings["qubit"]
-                if self.qubits[qubit].flux:
-                    qf_channel = self.qubits[qubit].flux.name
-                else:
-                    qf_channel = self.settings["qubit_channel_map"][qubit][2]
-                sequence.add(
-                    FluxPulse(
-                        start + pulse_settings["relative_start"], qf_duration, qf_amplitude, qf_shape, qf_channel, qubit
-                    )
-                )
-            elif pulse_settings["type"] == "virtual_z":
-                virtual_z_phases[pulse_settings["qubit"]] += pulse_settings["phase"]
-            else:
-                raise NotImplementedError(
-                    "Implementation of CZ gates using pulses of types other than `qf` or `virtual_z` is not supported yet."
-                )
-
-        return sequence, virtual_z_phases
+        return self.two_qubit_natives[frozenset(qubits)]["CZ"].sequence(start)
 
     def create_MZ_pulse(self, qubit, start):
-        pulse_kwargs = self.native_single_qubit_gates[qubit]["MZ"]
-        ro_duration = pulse_kwargs["duration"]
-        ro_frequency = pulse_kwargs["frequency"]
-        ro_amplitude = pulse_kwargs["amplitude"]
-        ro_shape = pulse_kwargs["shape"]
-        if self.qubits[qubit].readout:
-            ro_channel = self.qubits[qubit].readout.name
-        else:
-            ro_channel = self.settings["qubit_channel_map"][qubit][0]
-        return ReadoutPulse(start, ro_duration, ro_amplitude, ro_frequency, 0, ro_shape, ro_channel, qubit=qubit)
+        return self.qubits[qubit].native_gates.MZ.pulse(start)
```

Comment:
Maybe there is a way to make a single `create_pulse` (and a single `create_pulse_sequence`), since you just need one further argument to specify the kind of the gate.

Something like:
```py
def create_pulse(self, gate: Gate, qubit, **kwargs):
    natives = self.qubits[qubit].native_gates
    return getattr(natives, gate.name).pulse(**kwargs)
```